### PR TITLE
Fix the typing annotation for the return of parse()

### PR DESCRIPTION
**Pull request by rodsenra**
Basically a bug fix PR based on discussion for Issue https://github.com/sdispater/pendulum/issues/450
This PR incorporates a suggestion from @shot in
https://github.com/sdispater/pendulum/pull/320/files#diff-d79c6f58aeecd540ddd61603df9184aaR17

### CODE REVIEW

**Comment by bryanforbes** [comment]
@sdispater this looks good

File: pendulum/parser.py [hunk 1]
```diff
@@ -16,7 +19,9 @@
     CDuration = None
 
 
-def parse(text, **options):  # type: (str, **typing.Any) -> str
+def parse(
+    text, **options
+):  # type: (str, **typing.Any) -> typing.Union[Date, Time, Duration]
```
Comment (traviscook21) [inline pendulum/parser.py:24]: `DateTime` also possible, right?

**Comment by traviscook21** [comment]
I have code like this all of over some repos I'm working on
```
dt: pendulum.DateTime = pendulum.parse("2019-12-31")  # type: ignore
```

I wish it could be as simple as
```
dt = pendulum.parse("2019-12-31")
```

Anything I can do to help get this merged / released?

**Comment by sdispater** [comment]
Thanks for your contribution!

**Pull request merged**